tests_against_server/test_loops.py

```diff
@@ -343,8 +349,34 @@ def prog_maker(logspace_param):
         job = simulate_program_and_return(cfg, prog_maker(param))
         job.result_handles.wait_for_all_values()
         a_qua = job.result_handles.get("a").fetch_all()["value"]
-        a_list = get_equivalent_log_array(
-            np.round(np.logspace(param[0], param[1], param[2]))
-        )
+        a_list = get_equivalent_log_array(np.round(np.logspace(param[0], param[1], param[2])))
         assert len(a_list) == len(a_qua)
         assert np.allclose(a_list, a_qua, atol=1e-4)
+
+
+def test_qua_logspace_error(config):
+    def prog_maker(logspace_param):
+        with program() as prog:
+            t = declare(int)
+            t_st = declare_stream()
+            with for_(*qua_logspace(t, logspace_param[0], logspace_param[1], logspace_param[2])):
+                play("readout", "resonator", duration=t)
+                save(t, t_st)
+            with stream_processing():
+                t_st.save_all("a")
+        return prog
+
+    cfg = deepcopy(config)
+    arange_param_list = [
+        [np.log10(4), np.log10(10000), 60],
+    ]
+    for param in arange_param_list:
+        print(f"Test qua_list with {param}:")
+        try:
+            simulate_program_and_return(cfg, prog_maker(param))
+            raise Exception("Didn't catch error!")

```

Comment:
It looks like you're trying to assert that an error occurred due to a bad parameter. There is a neat way to do this using pytest. Let's say you expect a `ValueError`. You can run:

```
with pytest.raises(ValueError):
   simulate_program_and_return(cfg, prog_maker(param))
```

Not only is it neater, but it allows you to specify the exact error you expect. Often I've used this and it's been able to catch my code raising a _different_ error for some completely other reason than the expected one.

Reply:
Very nice, I just didn't know about it :)
Feel free to modify it if it is not too complicated

Reply:
It's fine for now since it works for whoever tested. it. But at least you can use it in the future

Reply:
I updated it just to get the gist of how it works, thanks!